// Copyright 2018 Iguazio
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package mlrun;

option go_package = "proto/build/log_collector";

import "google/protobuf/empty.proto";

service LogCollector {
  rpc StartLog(StartLogRequest) returns (StartLogResponse) {}
  rpc GetLog(GetLogRequest) returns (GetLogResponse) {}
}

message StartLogRequest {
  string runId = 1;
  string selector = 2;
}

message StartLogResponse {
  bool success = 1;
  string error = 2;
}

message GetLogRequest {
  string runId = 1;
  uint64 offset = 2;
  uint64 size = 3;
}

message GetLogResponse {
  bool success = 1;
  string error = 2;
  bytes log = 3;
}
